### ISSUE TITLE
Fix failing IE11 tests

### ISSUE THREAD

**Issue by fatso83** [sinonjs/samsam#42]
As detailed in https://github.com/sinonjs/samsam/pull/41#issuecomment-423980994 - and several other issues, the test suite currently has 7 failing tests in IE11. These are not all new - we had 4 failing prior to #38. 

Since IE11 is among our supported ES5 browsers in Sinon we should aim to fix these.

**Comment by mgred** [comment]
```
1) deepEqual
       returns false unequal errors:
     AssertionError: [assert.isFalse] Expected true to be false
      at referee.fail (node_modules/@sinonjs/referee/lib/referee.js:194)
      at ctx.fail (node_modules/@sinonjs/referee/lib/referee.js:84)
      at assertion (node_modules/@sinonjs/referee/lib/referee.js:92)
      at referee[type][name] (node_modules/@sinonjs/referee/lib/referee.js:112)
      at Anonymous function (lib/deep-equal.test.js:318)
```
When checking `Error` objects with [`deepEqual`](https://github.com/sinonjs/samsam/blob/master/lib/deep-equal.js#L90), we compare their stacks.

There is a note in [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/stack) regarding the `stack` property:
> Different browsers set this value at different times. For example, Firefox sets it when creating an Error object, while PhantomJS sets it only when throwing the Error, and MSDN docs also seem to match the PhantomJS implementation.

Since we create the error objects in the [test](https://github.com/sinonjs/samsam/blob/master/lib/deep-equal.test.js#L314) but never throw them, their `stack` property is `undefined`, I guess and therefore they are equal in IE 11.

I keep investigating on this issue and update this.

**Comment by mgred** [comment]
```
  4) getClassName
       returns the class name of an instance:
     AssertionError: [assert.equals] undefined expected to be equal to TestClass
      at referee.fail (node_modules/@sinonjs/referee/lib/referee.js:194)
      at ctx.fail (node_modules/@sinonjs/referee/lib/referee.js:84)
      at assertion (node_modules/@sinonjs/referee/lib/referee.js:92)
      at referee[type][name] (node_modules/@sinonjs/referee/lib/referee.js:112)
      at Anonymous function (lib/get-class-name.test.js:14)
  5) getClassName
       returns 'Object' for {}:
     AssertionError: [assert.equals] undefined expected to be equal to Object
      at referee.fail (node_modules/@sinonjs/referee/lib/referee.js:194)
      at ctx.fail (node_modules/@sinonjs/referee/lib/referee.js:84)
      at assertion (node_modules/@sinonjs/referee/lib/referee.js:92)
      at referee[type][name] (node_modules/@sinonjs/referee/lib/referee.js:112)
      at Anonymous function (lib/get-class-name.test.js:19)
```

Again [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name) demystifies. The `name` property is not supported in IE 11

**Comment by mroderick** [comment]
This has been fixed in #43 which was published as `@sinonjs/samsam@2.1.2`
